src/openmrs-backend-dependencies.ts

```diff
@@ -0,0 +1,92 @@
+import { openmrsFetch } from "@openmrs/esm-api";
+import * as semver from "semver";
+import { difference, isEmpty } from "lodash-es";
+
+var installedBackendModules: any[] = [];
+
+const originalOnload = System.constructor.prototype.onload;
+
+System.constructor.prototype.onload = function(err, id, deps) {
+  if (!err) {
+    System.import(id).then(module => {
+      console.log(module);
+      if (module.backendDependencies) {
+        checkBackendDeps(module.backendDependencies);
+      }
+    });
+  }
+  return originalOnload.apply(this, arguments);
+};
+
+function checkBackendDeps(backendModules: any) {
+  if (!isEmpty(installedBackendModules)) {
+    fetchInstalledBackendModules().then(({ data }) => {
+      installedBackendModules = data.results;
+      checkIfModulesAreInstalled(backendModules);
+    });
+  } else {
+    checkIfModulesAreInstalled(backendModules);
+  }
+}
+
+function checkIfModulesAreInstalled(requiredBackendModules) {
+  const missingOpenmrsBackendModules = getMissingBackendModules(
+    requiredBackendModules
+  );
+  const modulesWithWrongVersionInstalled = getMismatchedVersions(
+    requiredBackendModules
+  );
+  console.error(
```

Comment:
👍 this is a good start. In a future PR, let's surface these errors into a new tab of the devtools. Right now devtools only has the "Import Map" tab. Let's add a "Backend modules" tab in the future

<img width="697" alt="image" src="https://user-images.githubusercontent.com/5524384/65981568-9f179300-e436-11e9-88e5-73450a2a3f60.png">